strings/strings.go

```diff
@@ -343,3 +349,29 @@ func CommonSuffix(input ...string) string {
 
 	return string(suffix)
 }
+
+// Truncate shortens a given input string based on provided options.
+// Parameters:
+// - input: the original string to truncate.
+// - opts: optional settings to specify truncation length and omission suffix.
+// If opts is nil or certain fields are unspecified, defaults are applied:
+// Length defaults to 12 and Omission defaults to "...".
+func Truncate(input string, opts *TruncateOptions) string {
+	length := 12
+	omission := "..."
+
+	if opts != nil {
+		if opts.Length > 0 {
+			length = opts.Length
+		}
+		if opts.Omission != "" {
+			omission = opts.Omission
+		}
+	}
+
+	if len(input) <= length {
+		return input
+	}
```

Comment:
We should move this to the top for early returns. also good to add an empty input check

Reply:
It will check `len(input) <= length` before updating length from the option in that case @shahzadhaider1